### PR TITLE
Sync develop changes 9/11 - 9/13 to hdf5_1_14

### PR DESCRIPTION
**Pull request by lrknox**

### CODE REVIEW

File: config/cmake/FindHDFS.cmake [hunk 1]
```diff
@@ -8,7 +8,7 @@
 #  HDFS_FOUND, whether HDFS is found.
 
 execute_process(COMMAND $ENV{HADOOP_HOME}/bin/hadoop version OUTPUT_VARIABLE Hadoop_VERSION
-             RESULT_VARIABLE Hadoop_RETURN)
+             RESULT_VARIABLE_VALUE Hadoop_RETURN)
```
Comment (byrnHDF) [inline config/cmake/FindHDFS.cmake]: Should not be changed - "VALUE" is incorrect

**Review by byrnHDF — APPROVED** [review]

File: release_docs/RELEASE.txt [hunk 2]
```diff
@@ -47,6 +47,24 @@ New Features
...
+          CMake:      HDF5_ENABLE_THREADS (ON/OFF) (Default: ON)
+          Autotools:  --enable-threads    (yes/no) (Default: yes)
+
+      This option enables support for threading and concurrency algorithms
+      within the HDF5 library.  It is required for, but separate from, the
+      'threadsafe' configure option, which makes the HDF5 API safe to call from
+      multiple threads.  It is possible to enable the 'threads' option and
+      disable the 'threadsafe' option, but not vice versa.  The 'threads' option
+      must be on to enable the subfiling VFD.
+
```
Comment (derobins) [inline release_docs/RELEASE.txt]: This should not have been merged to 1.14. --enable-threads and HDF5_ENABLE_THREADS should be develop-only.

**Review by derobins — CHANGES_REQUESTED** [review]
Need to remove the comment about --enable-threads (and check to make sure that was not merged)

**Comment by lrknox** [comment]
> Need to remove the comment about --enable-threads (and check to make sure that was not merged)

Release note about --enable-threads has been removed.

**Review by byrnHDF — APPROVED** [review]

**Review by derobins — APPROVED** [review]

**Pull request merged**